src/reducers/setCursor.ts

```diff
@@ -71,7 +72,9 @@ const setCursor = (state: State, {
   //   }
   // })
 
-  const expanded = expandThoughts({ ...state, contextViews: newContextViews }, thoughtsResolved)
+  const expanded = expandThoughts({ ...state, contextViews: newContextViews }, thoughtsResolved, {
+    suppressExpansion: globals.suppressExpansion
+  })
```

Comment:
In`setCursor`,  `expandThoughts`is used to calculate the expanded state, which will actually be reflected in the view. In this case, global suppression should suppress the current cursor expansion.

However, in places like `expandBottom` , `expandThoughts` is used to know what will be the expanded state if the `expansionBasePath` is other than the current `cursor`.  Since this use case doesn't affect the view, global preference of suppression should not affect it.

So it made sense to pass the option to suppress the given `expansionBasePath` instead of directly using global preference.  It depends on what use case the `expandThoughts` is being called to pass the option. `setCursor` is using it with the current `cursor` as `expansionBasePath` which will directly be reflected in the view. That is why it needs to know about the global preference if the current cursor in the view should suppress or not, and pass it as the option to `expandThoughts`.

Reply:
In`setCursor`,  `expandThoughts`is used to calculate the expanded state, which will actually be reflected in the view. In this case, global suppression should suppress the current cursor expansion.

However, in places like `expandBottom` , `expandThoughts` is used to know what will be the expanded state if the `expansionBasePath` is other than the current `cursor`.  global preference of suppression should not affect it.

So it made sense to pass the option to suppress the given `expansionBasePath` instead of directly using global preference.  It depends on what use case the `expandThoughts` is being called to pass the option. `setCursor` is using it with the current `cursor` as `expansionBasePath` which will directly be reflected in the view. That is why it needs to know about the global preference if the current cursor in the view should suppress or not, and pass it as the option to `expandThoughts`.

Reply:
Thank you for the explanation! I figured there was some other reason.

What do you think a global vs a reducer here? Is my concern over performance unfounded?

Reply:
> What do you think a global vs a reducer here?

It can be both in my opinion. The only reason I see it needs to be global is the concern of re-render but some related reducers like `cursorPrev` always would be called along side with setting the `suppressExpansion`.